packages/api-key/src/models/api-key.ts

```diff
@@ -1,33 +1,68 @@
-import { generateEntityId } from "@medusajs/utils"
+import {
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 
 import {
   BeforeCreate,
   Entity,
   OnInit,
   PrimaryKey,
   Property,
+  Enum,
 } from "@mikro-orm/core"
 
-// TODO:
+const TypeIndex = createPsqlIndexStatementHelper({
+  tableName: "api_key",
+  columns: "type",
+})
+
 @Entity()
 export default class ApiKey {
   @PrimaryKey({ columnType: "text" })
   id: string
 
+  @Property({ columnType: "text" })
+  token: string
+
+  @Property({ columnType: "text" })
+  salt: string
+
+  @Property({ columnType: "text" })
+  redacted: string
+
+  @Property({ columnType: "text" })
+  title: string
+
+  @Property({ columnType: "text" })
+  @Enum({ items: ["publishable", "secret"] })
```

Comment:
I am not using the enum here as I think we should not couple the public interface with the DB model. What do others think?

Reply:
Fine by me. However, if the enums are the same in the public interface and the model, I guess it doesn't hurt to import it from utils and use it here?

Reply:
It really boils down to decoupling the database layer from the "API" layer. Accidentally, these two hold the same values, but we might want to rename them to something else externally (eg. `master` vs `secret`), but we won't necessarily want to change how the value is stored in the DB (a common scenario when you want to migrate data, and you want to do it one step at a time). In that case, you will mistakenly start saving `master` (since the enum will change) in the DB when you didn't really intend to do so, resulting in a bug.

If you are familiar with onion-layer architecture, this is the typical scenario where you want to decouple dependencies between your public-facing API and your database model